File: src/main/java/com/micabytes/WorldInterface.java
package com.micabytes;

@SuppressWarnings("MarkerInterface")
public interface WorldInterface {
  // This is simply a Marker interface with no methods or functions; created so that it is possible
  // to retrieve the game World object from the GameApplication without needing to put game
  // specific code into this library.
  void report(String msg, int msgTime);
}

File: src/main/java/com/micabytes/WorldInterface.kt
package com.micabytes

import com.micabytes.util.RpgUnit

interface WorldInterface {
  // This is simply a Marker interface with no methods or functions; created so that it is possible
  // to retrieve the game World object from the Game without needing to put game specific code into
  // this library.

  abstract fun get(id: String): RpgUnit?

  fun showMessage(msg: String, msgTime: Int)
}
